shell/common/crash_keys.cc

```diff
@@ -93,9 +95,17 @@ static_assert(kMaxCrashKeyNameLength <= crashpad::Annotation::kNameMaxLength,
 void SetCrashKey(const std::string& key, const std::string& value) {
   // Chrome DCHECK()s if we try to set an annotation with a name longer than
   // the max.
-  // TODO(nornagon): warn the developer (via console.warn) when this happens.
-  if (key.size() >= kMaxCrashKeyNameLength)
+  if (key.size() >= kMaxCrashKeyNameLength) {
+    node::Environment* env =
+        node::Environment::GetCurrent(JavascriptEnvironment::GetIsolate());
+    EmitWarning(env,
+                "The crash key name, \"" + key + "\", is longer than " +
+                    std::to_string(kMaxCrashKeyNameLength) +
+                    " bytes, ignoring it.",
+                "electron");
     return;
+  }
+
```

Comment:
Manual control, analysis, time of week, disabling auto bindings gradually. 🅰️ℹ️💯💯💯